server/docs/design/block-verification.md

```diff
@@ -0,0 +1,194 @@
+# Block Verification Design
+
+## Table of Contents
+1. [Purpose](#purpose)
+1. [Goals](#goals)
+1. [Terms](#terms)
+1. [Entities](#entities)
+1. [Design](#design)
+1. [Interfaces](#interfaces)
+1. [Enums](#enums)
+1. [Metrics](#metrics)
+1. [Exceptions](#exceptions)
+
+## Purpose
+The purpose of the Block Verification feature is to ensure that blocks received from consensus nodes are valid and have not been tampered with. This is achieved by re-calculating the block hash and verifying it against the signature provided by the consensus node.
+
+## Goals
+1. The block-node must re-create the block hash from the block items and verify that it matches the hash implied by the signature. 
+1. If verification fails, the block should be considered invalid, and appropriate error-handling procedures must be triggered.
+
+
+## Terms
+- Consensus Node (CN): A node that produces and provides blocks.
+- Block Items: The block data pieces (header, events, transactions, transaction result, state changes, proof) that make up a block.
+- Block Hash: A cryptographic hash representing the block’s integrity.
+- Signature: The cryptographic signature of the block hash created by Network private key (aka: LedgerId)
+- Public Key: The public key of the LedgerId that signed the block.
```

Comment:
This is another place where a definition list is probably clearer than the bullet list.

Reply:
Indeed it makes sense, it should be done! 

✅ Done!